src/components/tools/media/Homepage.js

```diff
@@ -1,57 +1,66 @@
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
-import React from 'react';
-import { injectIntl, FormattedMessage } from 'react-intl';
 import { connect } from 'react-redux';
-import { Grid, Row, Col } from 'react-flexbox-grid/lib';
-import ResourceList from './ResourceList';
-
-const localMessages = {
-  title: { id: 'media.intro.title', defaultMessage: 'Media Data' },
-  subtitle: {
-    id: 'media.intro.subtitle',
-    defaultMessage: 'What is Media Data?',
-  },
-  description: {
-    id: 'media.intro.description',
-    defaultMessage:
-      'MediaData uses a combination of machine learning data platforms and human research to bring you data, reports and analyses related to African media ecosystems',
-  },
-  loginTitle: {
-    id: 'media.intro.login.title',
-    defaultMessage: 'Have an Account? Login Now',
-  },
-};
+import { injectIntl } from 'react-intl';
+import { BlockRenderer } from '../../../lib/cmsUtils/blockRenderer';
+import { APP_TOOLS } from '../../../config';
+import { fetchPageContent } from '../../../actions/cmsActions';
+import CMSError from '../../common/CMSError';
+import { addNotice } from '../../../actions/appActions';
+import { LEVEL_ERROR } from '../../common/Notice';
 
-const Homepage = () => (
-  <div className="homepage">
-    <Grid>
-      <Row className="media-hero">
-        <Col lg={1} />
-        <Col lg={12}>
-          <h1>
-            <FormattedMessage {...localMessages.subtitle} />
-          </h1>
-          <p className="intro">
-            <FormattedMessage {...localMessages.description} />
-          </p>
-        </Col>
-      </Row>
-    </Grid>
-    <ResourceList />
-  </div>
-);
+
+const Homepage = (props) => {
+  const { getPageContent, route, pageData, fetchStatus } = props;
+
+  useEffect(() => {
+    getPageContent(APP_TOOLS, route?.path);
+  }, [getPageContent]);
+
+  if (fetchStatus === 'FETCH_FAILED') {
+    return (
+      <CMSError message="Failed to fetch content, please refresh your page" />
+    );
+  }
+
+  return (
+    <div className="homepage">
+      {
+      pageData?.blocks && pageData?.blocks.map((block, index) => (
+        <div key={index} className="mb-6">
+          <BlockRenderer
+            block={block}
+          />
+        </div>
+    ))
+    }
+    </div>
+  );
+};
 
 Homepage.propTypes = {
   intl: PropTypes.object.isRequired,
   // from context
   location: PropTypes.object.isRequired,
   params: PropTypes.object.isRequired, // params from router
+  route: PropTypes.object.isRequired,
+  getPageContent: PropTypes.func,
   // from state
+  pageData: PropTypes.object,
+  fetchStatus: PropTypes.string,
 };
 
-const mapStateToProps = () => ({});
+const mapStateToProps = (state) => ({
+  pageData: state.cms.pages.content.media,
+  fetchStatus: state.cms.pages.fetchStatus,
+});
 
-const mapDispatchToProps = () => ({});
+const mapDispatchToProps = (dispatch, ownProps) => ({
```

Comment:
I just thought `mapDispatchToProps` is a bit outdated and was intended to be used by class components. Since this is a Function Component, why not use [useDispatch](https://react-redux.js.org/api/hooks#usedispatch) inside the component.

Reply:
I hear you @koechkevin.  I decided to stick with the existing patterns in the current codebase to maintain consistency (you know, as they say 'when in Rome' 🫠).